crates/core/src/util/lending_pool.rs

```diff
@@ -58,31 +87,9 @@ impl<T> LendingPool<T> {
         let acq = self.sem.clone().acquire_owned();
         let pool_inner = self.inner.clone();
 
-        let queue_len = WORKER_METRICS.instance_queue_length.with_label_values(&db);
-        let queue_len_max = WORKER_METRICS.instance_queue_length_max.with_label_values(&db);
-        let queue_len_histogram = WORKER_METRICS.instance_queue_length_histogram.with_label_values(&db);
-
-        queue_len.inc();
-        let new_queue_len = queue_len.get();
-        queue_len_histogram.observe(new_queue_len as f64);
-
-        let mut guard = MAX_QUEUE_LEN.lock().unwrap();
-        let max_queue_len = *guard
-            .entry(db)
-            .and_modify(|max| {
-                if new_queue_len > *max {
-                    *max = new_queue_len;
-                }
-            })
-            .or_insert_with(|| new_queue_len);
-
-        drop(guard);
-        queue_len_max.set(max_queue_len);
-
         async move {
+            let _guard = QueueMetric::inc(db);
             let permit = acq.await.map_err(|_| PoolClosed)?;
```

Comment:
I'm fairly confident we should not have any more issues with this metric not resetting, or this metric going negative, with this scope guard in place.